File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/BuildExecutionDeploymentSettings.java
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment.compiler.CompilerSettings;
import org.immutables.value.Value;

import java.util.List;
import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableBuildExecutionDeploymentSettings.class)
public interface BuildExecutionDeploymentSettings {

  Optional<CompilerSettings> compiler();

  List<RequiredPlugin> requiredPlugins();

}

File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/BuildExecutionDeploymentSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment

import com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment.build_tools.BuildToolsSettings
import com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment.compiler.CompilerSettings

data class BuildExecutionDeploymentSettings(
    val buildTools: BuildToolsSettings? = null,
    val compiler: CompilerSettings? = null,
    val requiredPlugins: List<RequiredPlugin>? = null
)
